src/intervals/gff3/reader.jl

```diff
@@ -0,0 +1,70 @@
+
+type GFF3Reader <: Bio.IO.AbstractReader
+    state::Ragel.State
+    version::VersionNumber
+    sequence_regions::Vector{Interval{Void}}
+    key::StringField
+    save_directives::Bool
+    entry_seen::Bool
+    fasta_seen::Bool
+    unescape_needed::Bool
+
+    directive::StringField
+
+    preceding_directives::Vector{StringField}
+    preceding_directive_count::Int
+
+    directives::Vector{StringField}
+    directive_count::Int
+
+    function GFF3Reader(input::BufferedInputStream, save_directives::Bool=false)
+        return new(Ragel.State(gff3parser_start, input), VersionNumber(0), [],
+                   StringField(), save_directives, false, false, false,
+                   StringField(), StringField[], 0, StringField[], 0)
+    end
+end
+
+# GFF3 can end before the end of the file if there is a FASTA directive
+function Base.eof(reader::GFF3Reader)
+    return reader.state.finished || eof(reader.state.stream)
+end
+
+function Bio.IO.stream(reader::GFF3Reader)
+    return reader.state.stream
+end
+
+function Intervals.metadatatype(::GFF3Reader)
+    return GFF3Metadata
+end
+
+function Base.eltype(::Type{GFF3Reader})
+    return GFF3Interval
+end
+
+function GFF3Reader(input::IO, save_directives::Bool=false)
```

Comment:
I think it would be better to define `save_directive` as a keyword argument. Also, need a docstring here.